pkg/monitor/monitor/monitor_manager.go

```diff
@@ -447,22 +460,44 @@ func (m *MonitorManager) syncIngress(monitor *v1alpha1.TidbMonitor) error {
 }
 
 func (m *MonitorManager) syncPrometheusIngress(monitor *v1alpha1.TidbMonitor) error {
-	if monitor.Spec.Prometheus.Ingress == nil {
-		return m.removeIngressIfExist(monitor, prometheusName(monitor))
-	}
 
-	ingress := getPrometheusIngress(monitor)
-	_, err := m.deps.TypedControl.CreateOrUpdateIngress(monitor, ingress)
-	return err
+	shards := monitor.GetShards()
```

Comment:
Please add some UT and e2e cases for the changes.

Reply:
Ingress can‘t add unit test,because ingress use `TypedControlInterface` class create ,so we need to create a new PR, to  separate ingress implement.